File: app/src/main/java/com/example/theduckcardchatsystem/viewmodel/SettingsViewModel.java
package com.example.theduckcardchatsystem.viewmodel;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import com.example.theduckcardchatsystem.HomeActivity;
import com.example.theduckcardchatsystem.repository.SettingsRepository;
import com.google.firebase.database.DataSnapshot;

public class SettingsViewModel extends AndroidViewModel {
    private Application application;
    private SettingsRepository settingsRepository;
    private LiveData<DataSnapshot> dataSnapshotLiveData;
    private LiveData<DataSnapshot> photoSnapShotLiveData;
    private LiveData<DataSnapshot> changeNameLiveData;
    private LiveData<DataSnapshot> initUserLiveData;

    public SettingsViewModel(@NonNull Application application) {
        super(application);
        this.application = application;
        settingsRepository = new SettingsRepository(application);
        dataSnapshotLiveData = settingsRepository.dataSnapshotLiveData();
        photoSnapShotLiveData = settingsRepository.photoSnapShotLiveData();
        changeNameLiveData = settingsRepository.ChangeNameLiveData();
        initUserLiveData = settingsRepository.initUserLiveData();
    }
    public void setSettingsRepository(){
        settingsRepository.setSettingsData();
    }
    public void changePhotoViewModel(String path){
        settingsRepository.changePhoto(path);
    }
    public void changeName(String fullName){
        settingsRepository.ChangeName(fullName);
    }
    public void initUser(){
        settingsRepository.initUser();
    }
    public void changeUserName(String userName){
        settingsRepository.ChangeUserName(userName);
    }
    public void changeBio(String bio){
        settingsRepository.changeBio(bio);
    }
    public LiveData<DataSnapshot> dataSnapshotLiveData(){
        return  dataSnapshotLiveData;
    }
    public LiveData<DataSnapshot> photoSnapShotLiveData(){
        return photoSnapShotLiveData;
    }
    public LiveData<DataSnapshot> getInitUserLiveData(){
        return initUserLiveData;
    }

}

File: app/src/main/java/com/example/theduckcardchatsystem/viewmodel/SettingsViewModel.kt
package com.example.theduckcardchatsystem.viewmodel

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import com.example.theduckcardchatsystem.repository.SettingsRepository
import androidx.lifecycle.LiveData
import com.example.theduckcardchatsystem.repository.SingleChatRepository
import com.example.theduckcardchatsystem.adapters.SingleChatAdapter
import androidx.recyclerview.widget.RecyclerView
import com.google.firebase.database.DataSnapshot

class SettingsViewModel(private val application: Application) : AndroidViewModel(
    application
) {
    private val settingsRepository: SettingsRepository
    private val dataSnapshotLiveData: LiveData<DataSnapshot>
    private val photoSnapShotLiveData: LiveData<DataSnapshot>
    private val changeNameLiveData: LiveData<DataSnapshot>
    val initUserLiveData: LiveData<DataSnapshot>
    fun setSettingsRepository() {
        settingsRepository.setSettingsData()
    }

    fun changePhotoViewModel(path: String?) {
        settingsRepository.changePhoto(path)
    }

    fun changeName(fullName: String?) {
        settingsRepository.ChangeName(fullName)
    }

    fun initUser() {
        settingsRepository.initUser()
    }

    fun changeUserName(userName: String?) {
        settingsRepository.ChangeUserName(userName)
    }

    fun changeBio(bio: String?) {
        settingsRepository.changeBio(bio)
    }

    fun dataSnapshotLiveData(): LiveData<DataSnapshot> {
        return dataSnapshotLiveData
    }

    fun photoSnapShotLiveData(): LiveData<DataSnapshot> {
        return photoSnapShotLiveData
    }

    init {
        settingsRepository = SettingsRepository(application)
        dataSnapshotLiveData = settingsRepository.dataSnapshotLiveData()
        photoSnapShotLiveData = settingsRepository.photoSnapShotLiveData()
        changeNameLiveData = settingsRepository.ChangeNameLiveData()
        initUserLiveData = settingsRepository.initUserLiveData()
    }
}